src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -74,13 +74,15 @@
 
   private CloudSdk(Path sdkPath, String appCommandMetricsEnvironment,
```

Comment:
it's kind of bad form to have a constructor that allows the passing of contradictory arguments.